src/main/java/com/google/opensesame/projects/QueryFilter.java

```diff
@@ -0,0 +1,88 @@
+package com.google.opensesame.projects;
+
+import com.google.opensesame.util.ServletValidationException;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A Datastore filter stored in the form used by Objectify when loading entities from the Datastore.
+ *
+ * <p>See here for how Objectify conducts filtering on a query:
+ * https://www.javadoc.io/static/com.googlecode.objectify/objectify/6.0.6/com/googlecode/objectify/cmd/Query.html#filter(java.lang.String,java.lang.Object)
+ */
+class QueryFilter {
+  public static final String FILTER_QUERY_REGEX = "^[A-Za-z]+ (>|>=|!=|=|<|<=) .+$";
+
+  /**
+   * Parses a QueryFilter from a string and responds with errors to the provided servlet response
+   * object.
+   *
+   * <p>QueryFilter strings should be in the format "fieldName <comparator> value". Examples:
+   *
+   * <blockquote>
+   *
+   * <pre>
+   * "numMentors >= 2"
+   * "numInterestedUsers = 0"
+   * </pre>
+   *
+   * </blockquote>
+   *
+   * @param QueryFilterString The filter query string.
+   * @param response The servlet response to send errors to.
+   * @return Returns the QueryFilter or null if there was an error parsing the string.
```

Comment:
Yeah good catch; it seems I forgot to update the documentation after switching to throwing exceptions instead of returning null. I'll do a pass over all of the JavaDoc and comments.